### PR TITLE
Ports: Add x265, enable it in ffmpeg

### PR DESCRIPTION
**Pull request by danners**
Add x265 port, based on work by @tachyonray on Discord.

Compiles and runs, however when encoding a file in serenity, crashes somewhere in libpthread / kernel, so keep this as a draft.

`158.766 CrashReporter(58:59): 0x0a3ad01b: [libsystem.so] syscall1 +0xb (syscall.cpp:20 => syscall.cpp:19)
158.766 CrashReporter(58:59): 0x42cd97e2: [libc.so] futex +0x52 (syscall.h:30 => serenity.cpp:67)
158.766 CrashReporter(58:59): 0x2eaf069e: [libpthread.so] pthread_cond_timedwait.localalias +0x7e (serenity.h:42 => pthread_cond.cpp:101)
158.766 CrashReporter(58:59): 0x2eaf0722: [libpthread.so] pthread_cond_wait +0x12 (pthread_cond.cpp:84)
158.771 CrashReporter(58:59): 0x4f7931fe: [libx265.so.199] x265::FrameEncoder::getEncodedPicture(x265::NALList&) +0x4e ()
158.771 CrashReporter(58:59): 0x4f7d62ba: [libx265.so.199] x265::Encoder::encode(x265_picture const*, x265_picture*) +0x86a ()
158.771 CrashReporter(58:59): 0x4f7dbf71: [libx265.so.199] x265_encoder_encode +0x71 ()
158.798 CrashReporter(58:59): Generating backtrace took 20 ms
158.798 CrashReporter(58:59): --- Backtrace for thread #1 (TID 56) ---
158.798 CrashReporter(58:59): 0x4f99a72e: [libx265.so.199] dct32(short const*, short*, int) +0x3e ()
158.798 CrashReporter(58:59): 0x4fa45454: [libx265.so.199] x265::Quant::transformNxN(x265::CUData const&, unsigned char const*, unsigned int, short const*, unsigned int, short*, unsigned int, x265::TextType, unsigned int, bool) +0x3a4 ()
158.798 CrashReporter(58:59): 0x4f755703: [libx265.so.199] x265::Search::codeIntraLumaQT(x265::Mode&, x265::CUGeom const&, unsigned int, unsigned int, bool, x265::Search::Cost&, unsigned int const*) [clone .localalias] +0x373 ()
158.798 CrashReporter(58:59): 0x4f75d961: [libx265.so.199] x265::Search::estIntraPredQT(x265::Mode&, x265::CUGeom const&, unsigned int const*) +0x1151 ()
158.798 CrashReporter(58:59): 0x4f76a70e: [libx265.so.199] x265::Search::checkIntra(x265::Mode&, x265::CUGeom const&, x265::PartSize) +0xde ()
158.798 CrashReporter(58:59): 0x4f747c0a: [libx265.so.199] x265::Analysis::compressIntraCU(x265::CUData const&, x265::CUGeom const&, int) +0xa6a ()
158.803 CrashReporter(58:59): 0x4f7478e4: [libx265.so.199] x265::Analysis::compressIntraCU(x265::CUData const&, x265::CUGeom const&, int) +0x744 ()
158.803 CrashReporter(58:59): 0x4f751958: [libx265.so.199] x265::Analysis::compressCTU(x265::CUData&, x265::Frame&, x265::CUGeom const&, x265::Entropy const&) +0x678 ()
158.803 CrashReporter(58:59): 0x4f78bda6: [libx265.so.199] x265::FrameEncoder::processRowEncoder(int, x265::ThreadLocalData&) [clone .localalias] +0x566 ()
158.803 CrashReporter(58:59): 0x4f788d33: [libx265.so.199] x265::FrameEncoder::processRow(int, int) +0x73 ()
158.803 CrashReporter(58:59): 0x4fa16844: [libx265.so.199] x265::WaveFront::findJob(int) [clone .localalias] +0xb4 ()
158.803 CrashReporter(58:59): 0x4fa155ae: [libx265.so.199] x265::WorkerThread::threadMain() [clone .localalias] +0xfe ()
158.803 CrashReporter(58:59): 0x4f80e411: [libx265.so.199] x265_stack_align +0x21 ()
158.803 CrashReporter(58:59): 0x4fa15378: [libx265.so.199] x265::ThreadShim(void*) +0x28 ()
158.803 CrashReporter(58:59): 0x2eaef83f: [libpthread.so] pthread_create_helper +0x3f (pthread.cpp:52)
158.939 CrashReporter(58:59): Generating backtrace took 132 ms
`

### CODE REVIEW

**Comment by EWouters** [comment]
Was very cool to work on this in the discord voice chat. Just to clarify, the crash above happens when encoding a video with `ffmpeg` using the `x265` codec.

File: Ports/ffmpeg/package.sh [hunk 1]
```diff
@@ -17,6 +17,7 @@ configure() {
         --enable-cross-compile \
         --enable-gpl \
         --enable-libx264 \
+        --enable-libx265 \
```
Comment (timschumi) [inline Ports/ffmpeg/package.sh:20]: Make sure to also add `x265` to the dependencies.

Comment (danners) [reply Ports/ffmpeg/package.sh:20]: Thanks, added.

**Comment by timschumi** [comment]
So encoding and playing apparently works just fine on latest master (with #13915 applied and the `posix_memalign` patch removed), even if it is a little slow. The only thing that we need to fix is the fact that `x265` chooses a wrong installation directory, and installs itself to `<host path>/Build/<Arch>/Root/<host path>/Build/<Arch>/root/usr/local` instead.

**Comment by timschumi** [comment]
And, apparently, the solution to that was just to run `make install` without explicitly passing `DESTDIR`.

As far as I can tell the [`x265-new` branch](https://github.com/timschumi/serenity/commits/x265-new) in my repository seems to work just fine, so feel free to pick stuff over as you see fit.

**Comment by danners** [comment]
@timschumi I currently don't have access to a serenity computer. Since the memalign pr got merged I will close this then you can open a PR from your repo.

**Pull request closed without merging**